python_modules/dagster/dagster/components/core/defs_module.py

```diff
@@ -45,34 +44,33 @@ class ComponentFileModel(BaseModel):
     requirements: Optional[ComponentRequirementsModel] = None
 
 
-#########
-# MODULES
-#########
-
-
-@dataclass
-class DefsModuleComponent(Component):
-    path: Path
-
-    @classmethod
-    def from_context(cls, context: ComponentLoadContext) -> Optional["DefsModuleComponent"]:
-        """Attempts to load a component from the given context. Iterates through potential component
-        type matches, prioritizing more specific types: YAML, Python, plain Dagster defs, and component
-        folder.
-        """
-        # this defines the priority of the module types
-        module_types = (
-            WrappedYamlComponent,
-            WrappedPythonicComponent,
-            DagsterDefsComponent,
-            DefsFolderComponent,
+def get_component(context: ComponentLoadContext) -> Optional[Component]:
```

Comment:
should this be a method on `ComponentLoadContext` instead?

Reply:
Pretty clear this point we need to formalize and rationalize this because the ordering and naming is quite arbitrary.